From cee12914df0ee0747e0c8e4b4eb9c5643cf4126d Mon Sep 17 00:00:00 2001
From: Eric Sandeen <sandeen@redhat.com>
Date: Mon, 4 Aug 2014 19:00:57 -0400
Subject: [fs] btrfs: send: replace check with an assert in gen_unique_name

Message-id: <1407179109-13714-171-git-send-email-sandeen@redhat.com>
Patchwork-id: 87111
O-Subject: [RHEL7.1 PATCH 170/422] btrfs: send: replace check with an assert in gen_unique_name
Bugzilla: 1071179
RH-Acked-by: Zach Brown <zab@redhat.com>

From: David Sterba <dsterba@suse.cz>

commit 64792f253508268eb390a86f42f128d877b40776
Author: David Sterba <dsterba@suse.cz>
Date:   Mon Feb 3 18:24:09 2014 +0100

    btrfs: send: replace check with an assert in gen_unique_name

    The buffer passed to snprintf can hold the fully expanded format string,
    64 = 3x largest ULL + 3x char + trailing null.  I don't think that removing the
    check entirely is a good idea, hence the ASSERT.

    Signed-off-by: David Sterba <dsterba@suse.cz>
    Signed-off-by: Josef Bacik <jbacik@fb.com>
Signed-off-by: Eric Sandeen <sandeen@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/btrfs/send.c b/fs/btrfs/send.c
index 154a717..08edd0a 100644
--- a/fs/btrfs/send.c
+++ b/fs/btrfs/send.c
@@ -1418,11 +1418,7 @@ static int gen_unique_name(struct send_ctx *sctx,
  while (1) {
   len = snprintf(tmp, sizeof(tmp), "o%llu-%llu-%llu",
     ino, gen, idx);
-  if (len >= sizeof(tmp)) {
-   /* should really not happen */
-   ret = -EOVERFLOW;
-   goto out;
-  }
+  ASSERT(len < sizeof(tmp));
 
   di = btrfs_lookup_dir_item(NULL, sctx->send_root,
     path, BTRFS_FIRST_FREE_OBJECTID,
-- 
1.7.1